src/Proxy/DriveItemProxy.php

```diff
@@ -295,6 +296,54 @@ public function upload($name, $content, array $options = [])
 
         return new self($this->graph, $driveItem);
     }
+	
+	/**
+     * Creates a upload session to upload a file under this folder drive item.
+     *
+     * @param string $name
+     *        The name.
+     * @param string|resource|\GuzzleHttp\Psr7\Stream $content
+     *        The content.
+     * @param array $options
+     *        The options.
+     *
+     * @return UploadSessionProxy
+     *         The upload session created.
+     *
+     * @todo Support content type in options.
+     */
+    public function startUpload($name, $content, array $options = [])
+    {
+		$name         = rawurlencode($name);
+        $driveLocator = "/drives/{$this->parentReference->driveId}";
+        $itemLocator  = "/items/{$this->id}";
+		$endpoint = "$driveLocator$itemLocator:/$name:/createUploadSession";
+		
+		//Include extended options with existing item for upload session. Extended options to be overwritten if not mentioned
+		$item = [
+			'name' => isset($options['name']) ? $options['name'] : $name,
```

Comment:
I find it redundant to accept the file name from both `$name` and `$options`, please could you remove the latter?

Reply:
done 👍